dev/devicelab/lib/framework/ab.dart

```diff
@@ -41,32 +85,45 @@ class ABTest {
       ...summariesB.keys,
     };
 
-    final StringBuffer buffer = StringBuffer(
-      'Score\tAverage A (noise)\tAverage B (noise)\tSpeed-up\n',
-    );
-
+    final List<String> titles = <String>[
+      'Score',
+      'Average A', '(noise)',
+      'Average B', '(noise)',
+      'Speed-up'
+    ];
+    final List<FieldJustification> alignments = <FieldJustification>[
+      FieldJustification.LEFT,
+      FieldJustification.RIGHT, FieldJustification.LEFT,
+      FieldJustification.RIGHT, FieldJustification.LEFT,
+      FieldJustification.CENTER
+    ];
+
+    final List<int> lengths = List<int>.filled(6, 0);
+    accumulateLengths(lengths, titles);
     for (final String scoreKey in scoreKeyUnion) {
       final _ScoreSummary summaryA = summariesA[scoreKey];
       final _ScoreSummary summaryB = summariesB[scoreKey];
-      buffer.write('$scoreKey\t');
-
-      if (summaryA != null) {
-        buffer.write('${summaryA.average.toStringAsFixed(2)} (${_ratioToPercent(summaryA.noise)})\t');
-      } else {
-        buffer.write('\t');
-      }
-
-      if (summaryB != null) {
-        buffer.write('${summaryB.average.toStringAsFixed(2)} (${_ratioToPercent(summaryB.noise)})\t');
-      } else {
-        buffer.write('\t');
-      }
-
-      if (summaryA != null && summaryB != null) {
-        buffer.write('${(summaryA.average / summaryB.average).toStringAsFixed(2)}x\t');
-      }
+      accumulateLengths(lengths, <String>[
+        scoreKey,
+        summaryA?.averageString, summaryA?.noiseString,
+        summaryB?.averageString, summaryB?.noiseString,
+        summaryA?.improvementOver(summaryB),
+      ]);
+    }
 
-      buffer.writeln();
+    final StringBuffer buffer = StringBuffer();
+    alignments[0] = FieldJustification.CENTER;
+    formatResult(buffer, lengths, alignments, titles);
```

Comment:
Instead of splitting calling `accumulateLengths` and `formatResult` in 2 different `for` loops, it seems to be safer to first generate a 2D `List<List<String>> table`, and apply `accumulateLengths` and `formatResult` to the same `table`. That removes the possibility that in the future someone changes the content of one `for` loop, but forget to change that of another `for` loop.